packages/core/src/components/field-wrapper/helper-text-util.tsx

```diff
@@ -27,7 +27,13 @@ export function hasAnyText({
   validText?: string;
   helperText?: string;
 }) {
-  return invalidText || warningText || infoText || validText || helperText;
+  return (
+    invalidText?.trim() ||
+    warningText?.trim() ||
+    infoText?.trim() ||
+    validText?.trim() ||
+    helperText?.trim()
+  );
```

Comment:
Could possibly be simplified like that. Please check if this works
```suggestion
return [invalidText, warningText, infoText, validText, helperText]
  .map(text => text?.trim())
  .find(Boolean);
```